data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/validator/GetNewBlindedBlock.java

```diff
@@ -119,7 +119,7 @@ private static Function<SszData, SpecMilestone> getMilestoneSelector(final Spec
         return spec.getForkSchedule().getSpecMilestoneAtSlot(((BeaconBlock) sszData).getSlot());
       } else if (sszData instanceof BlindedBlockContents) {
         return spec.getForkSchedule()
-            .getSpecMilestoneAtSlot(((BlindedBlockContents) sszData).getBlock().getSlot());
+            .getSpecMilestoneAtSlot(((BlindedBlockContents) sszData).getSlot());
```

Comment:
maybe we could have a `Optional<BeaconBlock> fromSszData(sszData)` in `BeaconBlock` (and the corresponding for `BlindedBlockContents`) to confine the casting... just a nit

Reply:
And maybe the `toBeaconBlock`  and to(others) in `BlockContainer` to avoid `instanceof` in  `getResponseTypes`

Reply:
Yup, actually could be simplified even further since both instances are `BlockContainer`. Fixed in last commits.

Reply:
nice